src/mudlet.cpp

```diff
@@ -3744,6 +3744,103 @@ void mudlet::startAutoLogin(const QStringList& cliProfiles)
     }
 }
 
+void mudlet::handleTelnetUri(const QString& uri)
+{
+    // Parse telnet:// URI format: telnet://host[:port]
+    if (!uri.startsWith(qsl("telnet://"))) {
+        return;
+    }
+    
+    QString connectionString = uri.mid(9); // Remove "telnet://"
+    QString host;
+    int port = 23; // Default telnet port
+    
+    // Parse host and optional port
+    int colonIndex = connectionString.indexOf(':');
+    if (colonIndex != -1) {
+        host = connectionString.left(colonIndex);
+        bool ok;
+        port = connectionString.mid(colonIndex + 1).toInt(&ok);
+        if (!ok) {
+            port = 23; // Fall back to default if port parsing fails
+        }
+    } else {
+        host = connectionString;
+    }
+    
+    // Remove any trailing slashes or paths
+    int slashIndex = host.indexOf('/');
+    if (slashIndex != -1) {
+        host = host.left(slashIndex);
+    }
+    
+    if (host.isEmpty()) {
+        return;
+    }
+    
+    // Get list of existing profiles
+    QStringList profileList = QDir(getMudletPath(enums::profilesPath)).entryList(QDir::Dirs | QDir::NoDotAndDotDot, QDir::Name);
+    
+    // Check if any existing profile matches this server
+    QString matchingProfile;
+    QDateTime mostRecentTime;
+    
+    for (const QString& profileName : profileList) {
+        QString profileHost = readProfileData(profileName, qsl("url"));
+        QString profilePortStr = readProfileData(profileName, qsl("port"));
+        
+        bool ok;
```

Comment:
I'd suggest initialising it first
```suggestion
        bool ok = false;
```